drivers/tcs37727/tcs37727.c

```diff
@@ -0,0 +1,276 @@
+        return -1;
+    }
+
+    i2c_release(dev->i2c);
+
+    if (id != TCS37727_ID_VALUE) {
+        return -1;
+    }
+
+    return 0;
+}
+
+int tcs37727_init(tcs37727_t *dev, i2c_t i2c, uint8_t address, int atime_us)
+{
+    /* write device descriptor */
+    dev->i2c = i2c;
+    dev->addr = address;
+    dev->initialized = false;
+
+    i2c_acquire(dev->i2c);
+
+    /* initialize the I2C bus */
+    if (i2c_init_master(i2c, I2C_SPEED) < 0) {
+        i2c_release(dev->i2c);
+        return -1;
+    }
+
+    i2c_release(dev->i2c);
+
+    if (tcs37727_test(dev)) {
+        return -2;
+    }
+
+    i2c_acquire(dev->i2c);
+
+    if (i2c_write_reg(dev->i2c, dev->addr, TCS37727_CONTROL,
+                      TCS37727_CONTROL_AGAIN_4) != 1) {
+        i2c_release(dev->i2c);
+        return -3;
+    }
+    dev->again = 4;
+
+    if (i2c_write_reg(dev->i2c, dev->addr,
+                      TCS37727_ATIME_TO_REG(atime_us), 0) != 1) {
+        i2c_release(dev->i2c);
+        return -3;
+    }
+    dev->atime_us = atime_us;
+
+    dev->initialized = true;
+
+    i2c_release(dev->i2c);
+    return 0;
+}
+
+int tcs37727_set_rgbc_active(tcs37727_t *dev)
+{
+    uint8_t reg;
+
+    if (dev->initialized == false) {
+        return -1;
+    }
+
+    i2c_acquire(dev->i2c);
+    if (i2c_read_regs(dev->i2c, dev->addr, TCS37727_ENABLE, (char *)&reg, 1) != 1) {
+        i2c_release(dev->i2c);
+        return -1;
+    }
+
+    reg |= (TCS37727_ENABLE_AEN | TCS37727_ENABLE_PON);
+
+    if (i2c_write_reg(dev->i2c, dev->addr, TCS37727_ENABLE, reg) != 1) {
+        i2c_release(dev->i2c);
+        return -1;
+    }
+
+    i2c_release(dev->i2c);
+    return 0;
+}
+
+int tcs37727_set_rgbc_standby(tcs37727_t *dev)
+{
+    uint8_t reg;
+
+    if (dev->initialized == false) {
+        return -1;
+    }
+
+    i2c_acquire(dev->i2c);
+    if (i2c_read_regs(dev->i2c, dev->addr, TCS37727_ENABLE, (char *)&reg, 1) != 1) {
+        i2c_release(dev->i2c);
+        return -1;
+    }
+
+    reg &= ~TCS37727_ENABLE_AEN;
+    if (!(reg & TCS37727_ENABLE_PEN)) {
+        reg &= ~TCS37727_ENABLE_PON;
+    }
+
+    if (i2c_write_reg(dev->i2c, dev->addr, TCS37727_ENABLE, reg) != 1) {
+        i2c_release(dev->i2c);
+        return -1;
+    }
+
+    i2c_release(dev->i2c);
+    return 0;
+}
+
+static uint8_t tcs37727_trim_gain(tcs37727_t *dev, int rawc)
+{
+    uint8_t reg_again = 0;
+    int val_again = dev->again;
+
+    if (dev->initialized == false) {
+        return -1;
+    }
+
+    if (rawc < TCS37727_AG_THRESHOLD_LOW) {
+        switch (val_again) {
+            case 1:
+                reg_again = TCS37727_CONTROL_AGAIN_4;
+                val_again = 4;
+                break;
+
+            case 4:
+                reg_again = TCS37727_CONTROL_AGAIN_16;
+                val_again = 16;
+                break;
+
+            case 16:
+                reg_again = TCS37727_CONTROL_AGAIN_60;
+                val_again = 60;
+                break;
+
+            case 60:
```

Comment:
How about introducing a `break;` here to stay with the current gain value (`60`) while doing nothing. The `default` could stay with `return -1` for the error case